src/dune_engine/format_config.mli

```diff
@@ -10,7 +10,23 @@ module Language : sig
     | Dune
 end
 
-type t
+module Generic : sig
+  type 'files t
+
+  val field :
+       since:Dune_lang.Syntax.Version.t
+    -> default_files:'files
+    -> files:('files, Dune_lang.Decoder.fields) Dune_lang.Decoder.parser
+    -> 'files t option Dune_lang.Decoder.fields_parser
+
+  val equal : files:('files -> 'files -> bool) -> 'files t -> 'files t -> bool
+
+  val set_files : 'files1 t -> 'files2 -> 'files2 t
```

Comment:
I don't think we need to be that polymorphic:

```suggestion
  val set_files : 'files t -> 'files -> 'files t
```

Reply:
I think we do because at one point we want to use `set_files` to add a files field to a `unit t`. This piece of code can be found here in "src/dune_rules/super_context.ml", line 123.

Reply:
maybe we can make it `unit t -> 'a t -> 'a t` then. that ensures we don't use that to remove data